### PR TITLE
Add Google Analytics

### LINKED ISSUE #13: Add Google Analytics
Current Atom Plugin has GA, definitely want that here as well.

### PR DESCRIPTION
**Pull request by baylessj**
Closes #13.

Posts analytics info to the VSCode Google Analytics property in the SIGBots account (UA-84548828-7).

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -2,21 +2,36 @@ import * as vscode from "vscode";
...
-    vscode.window.showInformationMessage("Hello World from pros!")
-  );
+  analytics = new Analytics(context);
+
   const terminal = vscode.window.createTerminal("PROS Terminal");
   terminal.sendText("pros build-compile-commands");
 
-  vscode.commands.registerCommand("pros.upload&build", buildUpload);
+  vscode.commands.registerCommand("pros.upload&build", () => {
+    analytics.sendAction("build");
```
Comment (kunwarsahni01) [inline src/extension.ts:25]:
```suggestion
    analytics.sendAction("upload&build");
```

File: src/extension.ts [hunk 2]
```diff
@@ -2,21 +2,36 @@ import * as vscode from "vscode";
...
+    buildUpload();
+  });
 
   vscode.commands.registerCommand("pros.upload", upload);
```
Comment (kunwarsahni01) [inline src/extension.ts:29]:
```suggestion
  vscode.commands.registerCommand("pros.upload", () => {
    analytics.sendAction("upload");
    upload();
  });
```

File: src/extension.ts [hunk 3]
```diff
@@ -27,9 +42,13 @@ export function activate(context: vscode.ExtensionContext) {
 
   vscode.commands.registerCommand("pros.upgrade", upgradeProject);
```
Comment (kunwarsahni01) [inline src/extension.ts:43]:
```suggestion
  vscode.commands.registerCommand("pros.upgrade", () => {
    analytics.sendAction("upgrade");
    upgradeProject();
  });
```

**Review by kunwarsahni01 — CHANGES_REQUESTED** [review]
Want to have most actions tracked. Adding pageviews is great as well. LGTM

**Comment by baylessj** [comment]
@kunwarsahni01 This PR should be ready for a re-review from you whenever you get a chance :smile:

**Review by kunwarsahni01 — APPROVED** [review]
LGTM

**Pull request merged**